client/src/pages/answer/AnswerPage.tsx

```diff
@@ -176,10 +201,59 @@ const AnswerPage: React.FC<{ answerId: string }> = props => {
     ? `Files uploaded by ${displayName(differentUser)}`
     : 'Your current submission'
 
+  const menuItemList = result?.data.answer.versions.map(commit => (
+    <Menu.Item
+      key={commit.versionKey}
+      onClick={() =>
+        changeVersion({
+          variables: { id: props.answerId, versionID: commit.versionKey }
+        }).then(() => result.refetch())
+      }
+    >
+      {commit.commitMessage}
+    </Menu.Item>
+  ))
+  const menu = <Menu>{menuItemList}</Menu>
+
+  const dropdown = (
+    <Dropdown overlay={menu}>
+      <Button>{result.data.answer.currentVersionName}</Button>
+    </Dropdown>
+  )
+
+  const versioning = (
+    <div
+      style={{ display: 'flex', flexDirection: 'row', alignItems: 'center' }}
+    >
+      <div style={{ marginRight: '16px' }}>Submission version:</div>
+      <div style={{ minWidth: '20px' }}>{dropdown}</div>
+    </div>
+  )
+
+  const filesAndVersion = (
```

Comment:
I think this should be an extra and not in the title:

```jsx
const filesExtra = (
    <Row justify="space-between" align="middle" gutter={16}>
      <Col>{versioning}</Col>
      <Col>
        <VersionSaveComponent
          answerId={props.answerId}
          commitMessage={answer.currentVersionName}
          versionId={answer.currentVersionId}
        />
      </Col>
      <Col>
        <ArchiveDownload url={answer.sourceUrl}>
          Download source code
        </ArchiveDownload>
      </Col>
    </Row>
  )

[...]

extra={filesExtra}
```

Also use the ui framework (`antd`), please